### PR TITLE
Show Defold version in window title of main editor window

### LINKED ISSUE #7533: The title of the Defold Editor window contains a confusing version 2.0.
**Describe the bug (REQUIRED)**
It may or may not be a bug.

I know the story behind the 2.0, but on the other hand it hasn't been mentioned anywhere for a long time and people might get confused between the version in the window header and the version in the About window.

**To Reproduce (REQUIRED)**
Look at the Defold Editor window title.

**Expected behavior (REQUIRED)**
Title without any version?

**Defold version (REQUIRED):**
Defold 1.4.4

**Platforms (REQUIRED):**
macOS

### PR DESCRIPTION
**Pull request by britzl**
The main editor window title shows the confusing "Defold Editor 2.0" text. This version number has no correlation to the actual Defold version used. This has now been changed to instead show the actual Defold version used, eg "Defold Editor 1.4.5".

Fixes #7533 

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

**Review by matgis — CHANGES_REQUESTED** [review]
I don't like how the version number clutters up the title, and it's not really relevant enough to display prominently. Remember, this text can end up elsewhere where there is less room, like on the Windows task bar.
```
Defold Editor - 1.2.34 - My Game Project
```
I think we should just remove the "2.0".

**Comment by vlaaad** [comment]
Maybe "My Game - Defold 1.2.34" instead?

**Comment by britzl** [comment]
> Maybe "My Game - Defold 1.2.34" instead?

I like this!

**Comment by britzl** [comment]
> Remember, this text can end up elsewhere where there is less room, like on the Windows task bar

Also, remember that we put the game title first, without any regard to its length. We could potentially set a very long window title at any time.

File: editor/src/clj/editor/app_view.clj [hunk 1]
```diff
@@ -1680,7 +1680,7 @@ If you do not specifically require different script states, consider changing th
            (project/sub-select project-id active-resource-node sub-selection open-resource-nodes)))))))
 
 (defn- make-title
-  ([] "Defold Editor 2.0")
+  ([] (str "Defold " (System/getProperty "defold.version" "No version")))
```
Comment (matgis) [inline editor/src/clj/editor/app_view.clj]: Let's just have it say "Defold" if there is no version. Also, we can use the `(system/defold-version)` function.

```suggestion
  ([] (if-some [version (system/defold-version)]
        (str "Defold " version)
        "Defold"))
```

**Review by matgis — CHANGES_REQUESTED** [review]

File: editor/src/clj/editor/app_view.clj [hunk 2]
```diff
@@ -1680,7 +1680,7 @@ If you do not specifically require different script states, consider changing th
            (project/sub-select project-id active-resource-node sub-selection open-resource-nodes)))))))
 
 (defn- make-title
-  ([] "Defold Editor 2.0")
+  ([] (str "Defold " (System/getProperty "defold.version" "(No version)")))
```
Comment (matgis) [inline editor/src/clj/editor/app_view.clj:1683]: Let's just have it say "Defold" if there is no version. Also, we can use the `(system/defold-version)` function.

```suggestion
  ([] (if-some [version (system/defold-version)]
        (str "Defold " version)
        "Defold"))
```

**Review by matgis — CHANGES_REQUESTED** [review]

**Comment by vlaaad** [comment]
I think with the changes suggested by @matgis it's going to be nice and shippable!

**Pull request merged**